.github/actions/nix/setup_nix/action.yml

```
@@ -0,0 +1,235 @@
+name: "Setup Nix"
+description: "Setup Nix"
+inputs:
+  artifactory_user:
+    description: "The Artifactory user"
+    required: true
+  artifactory_password:
+    description: "The Artifactory password"
+    required: true
+  oss_only:
+    description: "Restrict upstream dependencies (e.g. Canton) to OSS versions (the equivalent of OSS_ONLY=1 in local checkouts)"
+    default: "false"
+    required: false
+  cache_version:
+    description: "The cache version"
+    required: true
+  should_save:
+    description: "If the nix cache should be saved"
+    # this should be run just from one job to ensure we avoid multi write conflicts, which makes everything worse
+    default: "false"
+  should_save_gcp:
+    description: "If the nix cache should be saved to the public GCP bucket"
+    default: "false"
+  upload_workload_identity_provider:
+    description: "The workload identity provider to use for uploading the cache"
+    required: false
+    default: ""
+  upload_service_account:
+    description: "The service account to use for uploading the cache"
+    required: false
+    default: ""
+
+runs:
+  using: "composite"
+  steps:
+    - name: Compute cache Key
+      id: cache_key
+      shell: bash
+      run: |
+        set -euxo pipefail
+        git ls-files nix/ | grep -v '[.]md$' | LC_ALL=C sort | xargs sha256sum -b > /tmp/nix-cache-key
+        uname -m >> /tmp/nix-cache-key # Add architecture to the cache key
+        echo "gh_cache_version: ${{ inputs.cache_version }}" >> /tmp/nix-cache-key # Add cache version to the cache key
+        if [ "${{ inputs.oss_only }}" == true ]; then
+          echo "Using OSS only dependencies"
+          echo "oss_only: ${{ inputs.oss_only }}" >> /tmp/nix-cache-key
+          touch /tmp/oss-only # Create a file to indicate that we are using OSS only dependencies (so we don't need to re-specifify oss_only to run_bash_command_in_nix)
+        fi
+        cat /tmp/nix-cache-key
+        cache_key=($(md5sum "/tmp/nix-cache-key"))
+        echo "cache_key=$cache_key" >> $GITHUB_ENV
+
+    - name: Download cache (for non-self-hosted)
+      if: ${{ !startsWith(runner.name, 'self-hosted') }}
+      shell: bash
+      run: |
+        set -euxo pipefail
+
+        if [ ${{ inputs.oss_only }} != 'true' ]; then
+          echo "Must use OSS only dependencies in GitHub-hosted runners"
+          exit 1
+        fi
+
+        echo "Latest nix cache:"
+
+        wget -q "https://storage.googleapis.com/splice-nix-cache-public/${cache_key}.tar.gz" -O cache.tar.gz || true
+        if [ ! -f "${cache_key}.tar.gz" ]; then
```

Matching the `wget`.

```suggestion
        if [ ! -s cache.tar.gz ]; then
```

---

doh, thanks